packages/instant/src/util/provider_factory.ts

```diff
@@ -29,6 +29,10 @@ export const providerFactory = {
         providerEngine.addProvider(new RPCSubprovider(rpcUrl));
         // // Start the Provider Engine
         providerEngine.start();
+        // This feels a bit dirty, but was the only way I could think of
+        // checking to see if this engine is our fallback engine, and not
+        // another Web3Provider engine provided by some dapp browser
+        (providerEngine as any).zeroExInstantFallbackEngine = true;
```

Comment:
I did this to ensure we are only reporting as the "Fallback" provider if the provider was constructed here.  This felt a bit dirty though.  

Is it possible that another dapp browser's constructor's name is "Web3ProviderEngine"? If so -- I think we need something like this to differentiate ours, unless someone can think of something better

Reply:
hmm we can just set the name of the providerState in `providerStateFactory.getInitialProviderStateFallback` to some constant instead of calling `envUtil.getProviderName(provider)`

Reply:
Good call -- done in 097bfe5